libraries/standard/mqtt/src/mqtt.c

```diff
@@ -163,7 +163,32 @@ MQTTStatus_t MQTT_Unsubscribe( MQTTContext_t * const pContext,
 
 MQTTStatus_t MQTT_Disconnect( MQTTContext_t * const pContext )
 {
-    return MQTTSuccess;
+    size_t packetSize;
+    int32_t bytesSent;
+
+    MQTTStatus_t status = MQTT_GetDisconnectPacketSize( &packetSize );
+
+    if( status == MQTTSuccess )
+    {
+        status = MQTT_SerializeDisconnect( &( pContext->networkBuffer ) );
+    }
+
+    if( status == MQTTSuccess )
+    {
+        bytesSent = sendPacket( pContext, packetSize );
+
+        if( bytesSent < 0 )
+        {
+            status = MQTTSendFailed;
```

Comment:
I'm wondering about MQTTStatus_t here. Should we have one Network Error status rather than send fail or receive fail.  The more specific error code can be communicated via context, (error code bubbling up, I'm still working on that proposal as part of transport layer ).